### ISSUE TITLE
NuxtLink no longer type-safe after `content.documentDriven` is configured

### ISSUE THREAD

**Issue by cayter** [victorgarciaesgi/nuxt-typed-router#115]
**Describe the bug**
NuxtLink no longer type-safe after `content.documentDriven` is configured.

**Expected behavior**
NuxtLink should still be type-safe after `content.documentDriven` is configured. It seems both `useContent()` and `useRouter().getRoutes()` can be used to generate the route types.

**Environnement infos**

```sh
------------------------------
- Operating System: Darwin
- Node Version:     v20.5.1
- Nuxt Version:     3.7.1
- CLI Version:      3.7.3
- Nitro Version:    2.6.2
- Package Manager:  bun@0.8.2
- Builder:          -
- User Config:      content, devtools, modules
- Runtime Modules:  @nuxt/content@2.8.2, @nuxt/image@1.0.0-rc.1, @nuxt/ui@2.8.0, @nuxtjs/color-mode@3.3.0, @nuxtjs/i18n@8.0.0-rc.4, nuxt-typed-router@3.3.1
- Build Modules:    -
------------------------------
```

**Your nuxt.config.ts**

```ts
import { resolve } from "node:path";

// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  content: {
    documentDriven: true,
    sources: {
      content: {
        driver: "fs",
        prefix: "/docs",
        base: resolve(__dirname, "content/docs"),
      },
    },
  },
  devtools: {
    enabled: true,
    timeline: {
      enabled: true,
    },
  },
  modules: [
    "@nuxt/content",
    "@nuxt/image",
    "@nuxt/ui",
    "@nuxtjs/color-mode",
    "@nuxtjs/i18n",
    "nuxt-typed-router",
  ],
});
```

**Comment by victorgarciaesgi** [comment]
Hi! Didn't know about this option i'll look at it
What does it do ?

**Comment by cayter** [comment]
https://content.nuxtjs.org/guide/writing/document-driven

I'm trying to co-locate my app and it's docs in the same Nuxt app. I just started digging into NuxtJS seriously since yesterday after finding it much more complete as compared with React's BYOS which could help me ship faster. So yeah I am not very sure what this implies.

**Comment by victorgarciaesgi** [comment]
Sorry for the late response, I have a fix, it will be published on 3.5.0

**Comment by victorgarciaesgi** [comment]
Fixed in `v3.5.0`
